discovery/service/service.py

```diff
@@ -110,14 +113,48 @@ def get_service_facts(input_context: InputContext, service: ServiceData, hosts:
 
         return response.get(hosts[0]).get("status")
 
+    @staticmethod
+    def _get_systemd_env_details(file_content) -> dict:
+        import configparser
+        env_dict = dict()
+        if not file_content:
+            return env_dict
+
+        configParser = configparser.RawConfigParser(dict_type=MultiOrderedDict, strict=False)
+        configParser.read_string(file_content)
+        try:
+            env_details = configParser.get(section='Service', option='Environment')
+            for line in env_details.splitlines():
+                key, value = line.split('=', maxsplit=1)
+                env_dict[key.strip('"')] = value.strip('"')
+        except Exception:
+            pass
+        return env_dict
+
+    @staticmethod
+    def get_service_environment_variable(input_context: InputContext, service: ServiceData, hosts: list) -> dict:
+        # Large values in service environment are getting truncated. As an alternate method, we will get
+        # it from service and override variables
+        env_map = dict()
+
+        service_facts = AbstractPropertyBuilder.get_service_facts(input_context, service, hosts)
+        service_file = service_facts.get('FragmentPath')
+        override_file = service_facts.get('DropInPaths')
+
+        content = ServicePropertyManager.slurp_remote_file(input_context=input_context, hosts=hosts, file=service_file)
+        env_map.update(AbstractPropertyBuilder._get_systemd_env_details(content.get(hosts[0])))
+
+        content = ServicePropertyManager.slurp_remote_file(input_context=input_context, hosts=hosts, file=override_file)
```

Comment:
is this called twice by design?

Reply:
Yes, we try to get the default service env variable and then override it by override file ( if present )